sdk/csharp/src/Dagger.SDK.CodeFixes/AddDefaultPathAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Dagger.SDK.Analyzers;
+
+[
+    ExportCodeFixProvider(
+        LanguageNames.CSharp,
+        Name = nameof(AddDefaultPathAttributeCodeFixProvider)
+    ),
+    Shared
+]
+public class AddDefaultPathAttributeCodeFixProvider : CodeFixProvider
+{
+    public sealed override ImmutableArray<string> FixableDiagnosticIds =>
+        [DiagnosticDescriptors.DirectoryParameterShouldHaveDefaultPath.Id];
+
+    public sealed override FixAllProvider GetFixAllProvider() =>
+        WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var root = await context
+            .Document.GetSyntaxRootAsync(context.CancellationToken)
+            .ConfigureAwait(false);
+
+        if (root == null)
+            return;
+
+        var diagnostic = context.Diagnostics.First();
+        var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+        var parameter = root.FindToken(diagnosticSpan.Start)
+            .Parent?.AncestorsAndSelf()
+            .OfType<ParameterSyntax>()
+            .FirstOrDefault();
+
+        if (parameter == null)
+            return;
+
+        context.RegisterCodeFix(
+            CodeAction.Create(
+                title: "Add [DefaultPath(\".\")] attribute",
+                createChangedDocument: c =>
+                    AddDefaultPathAttributeAsync(context.Document, parameter, c),
+                equivalenceKey: nameof(AddDefaultPathAttributeCodeFixProvider)
+            ),
+            diagnostic
+        );
+    }
+
+    private async Task<Document> AddDefaultPathAttributeAsync(

```

Comment:
I'm curious about that one: I'm not sure I understand the purpose of this CodeFix, why adding a default path on directory argument?

Reply:
e.g +defaultPath only works on `dagger.Directory` so this is to show the user they can add a default Path. It could also only be a hint.

Reply:
Oh okay, that's a great suggestion then!